model/src/main/kotlin/licenses/ResolvedLicenseInfo.kt

```diff
@@ -83,12 +84,20 @@ data class ResolvedLicenseInfo(
      * in any of the configured [LicenseFilePatterns] matched against the root path of the package (or project).
      */
     fun mainLicense(): SpdxExpression? {
-        val licenseMatcher = PathLicenseMatcher(LicenseFilePatterns.getInstance())
-        val licensePaths = flatMap { resolvedLicense ->
+        val licenseFilePatterns = LicenseFilePatterns.getInstance()
+        val fileMatcher = FileMatcher(licenseFilePatterns.allLicenseFilenames, ignoreCase = true)
+        val licenseMatcher = PathLicenseMatcher(licenseFilePatterns)
+
+        val relevantResolvedLicenses = mapNotNull { resolvedLicense ->
+            val locations = resolvedLicense.locations.filterTo(mutableSetOf()) { fileMatcher.matches(it.location.path) }
+            if (locations.isNotEmpty()) resolvedLicense.copy(locations = locations) else null
+        }
```

Comment:
If the filtering of resolved licenses against the paths was extracted to a function, and also some comment was added why filtering first is important for performance, it could be less likely the issue gets re-introduced. What do you think ?

Reply:
How does extracting the filtering to function help to avoid reintroducing the problem? Any user would still need to be aware of the function, and make use of it.

Reply:
If someone attempts to refactor the code within `mainLicense`, it currently is not obvious IMO.

Reply:
I decided to add a code comment instead.